File: src/lib/api/rustRcon.ts
/**
 * Typed wrappers around Tauri rust_rcon_* commands (Facepunch RCON / WebSocket).
 *
 * Backend: src-tauri/src/services/rust_rcon.rs (tokio-tungstenite WebSocket).
 */

import { invoke } from "@tauri-apps/api/core";

export const rustRconApi = {
  /** Connect ws://host:port/password and authenticate. Returns session key. */
  connect: (host: string, port: number, password: string) =>
    invoke<string>("rust_rcon_connect", { host, port, password }),

  /** Send a console command. Response arrives via `recv()`. */
  send: (key: string, cmd: string) =>
    invoke<void>("rust_rcon_send", { key, cmd }),

  /** Drain any messages buffered since last call. Polled from UI. */
  recv: (key: string) =>
    invoke<string[]>("rust_rcon_recv", { key }),

  disconnect: (key: string) =>
    invoke<void>("rust_rcon_disconnect", { key }),
};

File: src-tauri/src/commands/rust_rcon.rs
//! Tauri commands wrapping `services::rust_rcon`.

use tauri::State;
use crate::error::AppResult;
use crate::services::rust_rcon::{self, RustRconState};

#[tauri::command]
pub async fn rust_rcon_connect(
    state: State<'_, RustRconState>,
    host: String,
    port: u16,
    password: String,
) -> AppResult<String> {
    rust_rcon::connect(&state, &host, port, &password).await
}

#[tauri::command]
pub async fn rust_rcon_send(
    state: State<'_, RustRconState>,
    key: String,
    cmd: String,
) -> AppResult<()> {
    rust_rcon::send(&state, &key, &cmd).await
}

#[tauri::command]
pub async fn rust_rcon_recv(
    state: State<'_, RustRconState>,
    key: String,
) -> AppResult<Vec<String>> {
    rust_rcon::recv(&state, &key).await
}

#[tauri::command]
pub async fn rust_rcon_disconnect(
    state: State<'_, RustRconState>,
    key: String,
) -> AppResult<()> {
    rust_rcon::disconnect(&state, &key).await
}
